bittide-instances/src/Bittide/Instances/Hitl/IlaPlot.hs

```diff
@@ -443,51 +443,48 @@ callistoClockControlWithIla dynClk clk rst ccc IlaControl{..} mask ebs =
 
   -- collect all plot data
-  localData =
-    let rfStageChange CallistoResult{..} = case reframingState of
-          Detect {} -> ToDetect
-          Wait {}   -> ToWait
-          Done {}   -> ToDone
-
-        height = SNat @AccWindowHeight
-        idcs = unbundle (stability <$> result)
-
-        ebsC = ebsDiffCompress scheduledCapture <$> ebs
-
-        -- get the points in time where the monitored values change
-        stableUpdates  = changepoints clk rst enableGen <$> (fmap stable <$> idcs)
-        settledUpdates = changepoints clk rst enableGen <$> (fmap settled <$> idcs)
-        modeUpdate     = changepoints clk rst enableGen (rfStageChange <$> result)
-
-        combine eb stU seU ind = (,,)
-          <$> eb
-          <*> (orNothing <$> stU <*> (stable  <$> ind))
-          <*> (orNothing <$> seU <*> (settled <$> ind))
-
-        noChange = fromMaybe NoChange . maybeSpeedChange
-
-     in PlotData
-          <$> bundle (zipWith4 combine ebsC stableUpdates settledUpdates idcs)
-          <*> accWindow height clk rst enableGen (noChange <$> result)
-          <*> mux modeUpdate (rfStageChange <$> result) (pure Stable)
+  localData = PlotData
+    <$> bundle (zipWith4 combine ebsC stableUpdates settledUpdates idcs)
+    <*> accWindow height clk rst enableGen (noChange <$> result)
+    <*> mux modeUpdate (rfStageChange <$> result) (pure Stable)
+   where
+    rfStageChange CallistoResult{..} = case reframingState of
+      Detect {} -> ToDetect
+      Wait {}   -> ToWait
+      Done {}   -> ToDone
+
+    height = SNat @AccWindowHeight
+    idcs = unbundle (stability <$> result)
+
+    -- get the points in time where the monitored values change
+    stableUpdates  = changepoints clk rst enableGen <$> (fmap stable <$> idcs)
+    settledUpdates = changepoints clk rst enableGen <$> (fmap settled <$> idcs)
+    modeUpdate     = changepoints clk rst enableGen (rfStageChange <$> result)
+
+    combine eb stU seU ind = (,,)
+      <$> eb
+      <*> (orNothing <$> stU <*> (stable  <$> ind))
+      <*> (orNothing <$> seU <*> (settled <$> ind))
+
+    noChange = fromMaybe NoChange . maybeSpeedChange
 
   -- compress the elastic buffer data via only reporting the
   -- differences since the last capture
-  ebsDiffCompress = curry $
-    let transF lastDataCount (trigger, curDataCount) =
-          let diff = curDataCount - lastDataCount
-              half = extend @_
-                @(CompressedBufferSize - 1)
-                @(m - CompressedBufferSize + 1)
-                maxBound
-              truncDiff = truncateB @_
-                @CompressedBufferSize
-                @(m - CompressedBufferSize)
-                diff
-           in (, truncDiff)
-            $ bool lastDataCount curDataCount
-            $ trigger || abs diff > half
-     in mealyB clk rst enableGen transF (0 :: DataCount m)
+  (ebDataChange, ebsC) = second unbundle
+    $ mealyB clk rst enableGen transF (repeat 0) (scheduledCapture, bundle ebs)
```

Comment:
```suggestion
  (ebDataChange, unbundle -> ebsC) =
    mealyB clk rst enableGen transF (repeat 0) (scheduledCapture, bundle ebs)
```

A view pattern is more obvious than a HO combinator IMO. What do you think?

Reply:
It depends. The higher order use of `second` here results in a linear control flow, while view patterns introduce a branching flow. I'll personally prefer to use linear flow, as it is easier to read, copy paste (while coding), and extend with type annotations if necessary. I decided to use `second` here, as it is already imported anyway, so it comes at no cost of an extra import.